### PR TITLE
Make totalAmount a Bignum - Closes #797

### LINKED ISSUE #797: Schema totalAmount should perform the check with Bignum
RIght now, the schema check is useless. We need to change the totalAmount check in logic to use Bignum.

https://github.com/LiskHQ/lisk/blob/5694df5d734c9d5a5a4f32674ae57acfd476ded0/logic/transaction.js#L527

It should be completed with some unit tests addressing this part of the code.

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #797

### CODE REVIEW

File: helpers/constants.js [hunk 1]
```diff
@@ -93,7 +94,7 @@ var constants = {
 	},
 	signatureLength: 196,
 	// WARNING: When changing totalAmount you also need to change getBlockRewards(int) SQL function!
-	totalAmount: 10000000000000000,
+	totalAmount: new bignum(10000000000000000),
```
Comment (MaciejBaj) [inline helpers/constants.js]: Please check the [failing build](https://jenkins.lisk.io/job/lisk-core/view/change-requests/job/PR-847/) on Jenkins. There are still places in the code where we use `constants.totalAmount `expecting it to be a number, not a bignum.

File: logic/blockReward.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ BlockReward.prototype.calcSupply = function (height) {
 
 	if (height < this.rewardOffset) {
 		// Rewards not started yet
-		return constants.totalAmount;
+		return constants.totalAmount.toInteger();
```
Comment (4miners) [inline logic/blockReward.js]: According to documentation (https://github.com/MikeMcl/bignumber.js/) there is no `toInteger()`

**Comment by MaciejBaj** [comment]
As we discussed with @4miners - `amount` should be stored as a string. Every calculation and comparison should be done using `bignum.js` library and the result stored as a string once again. We can be sure then that the value will never be malformed because of exceeding `Number.MAX_SAFE_INTEGER`

**Comment by 4miners** [comment]
Will be solved as part of https://github.com/LiskHQ/lisk/issues/883

**Pull request closed without merging**